downstairs/src/lib.rs

```diff
@@ -1617,6 +1617,10 @@ impl ActiveConnection {
     ) -> Message {
         let upstairs_connection = self.upstairs_connection;
 
+        if flags.lossy && random() && random() {
```

Comment:
Nit: the `random() && random()` is non-obvious, maybe something like
```suggestion
        if flags.lossy && random_range(0..4) == 0 { // 25% failure rate
```